src/views/topics/GdpTopic06View.vue

```diff
@@ -1,17 +1,237 @@
 <template>
-    <TopicMeta :topic="topic" />
+  <TopicMeta :topic="topic" />
+  <div id="Pascal-Brainchon-wrapper">
+    <canvas id="Pascal-Brainchon-canvas" width="500" height="500" />
+  </div>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import { indexTopicMap } from '@/data';
 import { Topic } from '@/types';
+import { fabric } from 'fabric';
+import { makeLabel, makeLine, makeMovablePoint } from '@/utils/canvas';
+import { IEvent } from 'fabric/fabric-impl';
+import { calculateLineIntersectInLinearEquation, solveLinearEquation } from '@/utils/geometry';
+
 const topic = indexTopicMap.get(6) as Topic;
+
+/**
+ * Type definitions
+ */
+type Intersection = {
+  status?: string
+  points?: Array<fabric.Point>
+}
+
+type Circle = fabric.Circle & {
+  [key: string]: any,
+  intersects?: Array<Circle>,
+  upLine?: Array<fabric.Line>,
+  downLine?: Array<fabric.Line>,
+  crossLines?: Array<fabric.Line>,
+  label?: fabric.Text,
+}
+
+type Line = fabric.Line & {
+  p1?: Circle,
+  p2?: Circle,
+  m?: number,
+  b?: number,
+}
+
+type Coord = {
+  x: number,
+  y: number,
+}
+
+/**
+ * Utility functions
+ */
+const setLableToPoint = (labels: Array<fabric.Text>, points: Array<Circle>): void => {
```

Comment:
Why arrow function?

Reply:
Wrapping everything in this arrow function extracts common logic and simplifies code. For example, next I need to apply a unique offset on each label so that it doesn't stick to the lines and points. This would be the perfect place to do so.